lib/iris/cube.py

```diff
@@ -1632,7 +1638,38 @@ def dtype(self):
 
     @dtype.setter
     def dtype(self, dtype):
-        self._dtype = dtype
+        if dtype != self.dtype:
+            if dtype is not None:
+                if not self.has_lazy_data():
+                    emsg = 'Cube does not have lazy data, cannot set dtype.'
+                    raise ValueError(emsg)
+                dtype = np.dtype(dtype)
+                if dtype.kind != 'i':
+                    emsg = ('Can only cast lazy data to integral dtype, '
+                            'got {!r}.')
```

Comment:
This [numpy scalars link](http://numpy.readthedocs.io/en/latest/reference/arrays.scalars.html#) was handy ...

And this pattern seem to work:
```python
>>> integral = [np.arange(10, dtype=np.int8),
... np.arange(10, dtype=np.int16),
... np.arange(10, dtype=np.int32),
... np.arange(10, dtype=np.int64),
... np.arange(10, dtype=np.uint8),
... np.arange(10, dtype=np.uint16),
... np.arange(10, dtype=np.uint32),
... np.arange(10, dtype=np.uint64)]
>>> for i in integral:
...     print(isinstance(i[0], np.integer))
... 
True
True
True
True
True
True
True
True
>>> f = np.arange(10, dtype=float)
>>> isinstance(f, np.integer)
False
```